cs/src/core/Index/FASTER/FASTERThread.cs

```diff
@@ -465,66 +440,40 @@ internal async ValueTask<(Status, Output)> CompleteIOPendingReadRequestsAsync(lo
                 }
             }
 
-            if (opCtx.ioPendingRequests.TryGetValue(request.id, out PendingContext pendingContext))
+
+            ref Key key = ref pendingContext.key.Get();
+
+            OperationStatus internalStatus = InternalContinuePendingRead(opCtx, request, ref pendingContext, currentCtx);
+
+            request.Dispose();
+
+            Status status;
+            // Handle operation status
+            if (internalStatus == OperationStatus.SUCCESS || internalStatus == OperationStatus.NOTFOUND)
             {
-                ref Key key = ref pendingContext.key.Get();
+                status = (Status)internalStatus;
+            }
+            else
+            {
+                throw new Exception($"Unexpected {nameof(OperationStatus)} while reading => {internalStatus}");
+            }
 
-                // Remove from pending dictionary
-                opCtx.ioPendingRequests.Remove(request.id);
 
-                OperationStatus internalStatus;
-                // Issue the continue command
-                if (pendingContext.type == OperationType.READ)
-                {
-                    internalStatus = InternalContinuePendingRead(opCtx, request, ref pendingContext, currentCtx);
-                }
-                else
-                {
-                    internalStatus = InternalContinuePendingRMW(opCtx, request, ref pendingContext, currentCtx); ;
```

Comment:
We need to support pending RMW operations. Why is this deleted? RMW operations may either "retry" or go "pending".

Reply:
This code is only hit be readyAsync execution path.
Neither RWM or RMWAsync hit this, and all RMW (and other write operations) are unchanged.